crates/re_types_builder/src/codegen/docs/mod.rs

```diff
@@ -0,0 +1,170 @@
+use crate::codegen::common::ExampleInfo;
+use crate::objects::FieldKind;
+use crate::CodeGenerator;
+use crate::Object;
+use crate::ObjectField;
+use crate::ObjectKind;
+use crate::Objects;
+use crate::Reporter;
+use camino::Utf8PathBuf;
+use std::collections::BTreeSet;
+use std::fmt::Write;
+
+use super::common::get_documentation;
+
+macro_rules! putln {
+    ($o:ident) => {let _ = writeln!($o);};
+    ($o:ident, $($tt:tt)*) => {let _ = writeln!($o, $($tt)*);};
+}
+
+pub struct DocsCodeGenerator {
+    docs_dir: Utf8PathBuf,
+}
+
+impl DocsCodeGenerator {
+    pub fn new(docs_dir: impl Into<Utf8PathBuf>) -> Self {
+        Self {
+            docs_dir: docs_dir.into(),
+        }
+    }
+}
+
+impl CodeGenerator for DocsCodeGenerator {
+    fn generate(
+        &mut self,
+        _reporter: &Reporter,
+        objects: &Objects,
+        _arrow_registry: &crate::ArrowRegistry,
+    ) -> BTreeSet<camino::Utf8PathBuf> {
+        re_tracing::profile_function!();
+
+        let mut filepaths = BTreeSet::new();
+
+        let components = objects.ordered_objects(Some(ObjectKind::Component));
+        for object in objects.ordered_objects(Some(ObjectKind::Archetype)) {
+            // skip test-only archetypes
+            if object.name.starts_with("AffixFuzzer") {
+                continue;
+            }
+
+            let order = object.order;
+            let top_level_docs = get_documentation(&object.docs, &[]);
+            let examples = object
+                .docs
+                .tagged_docs
+                .get("example")
+                .iter()
+                .flat_map(|v| v.iter())
+                .map(ExampleInfo::parse)
+                .collect::<Vec<_>>();
+
+            let mut o = String::new();
+
+            frontmatter(&mut o, &object.name, order);
+            putln!(o);
+            for mut line in top_level_docs {
+                if line.starts_with(char::is_whitespace) {
+                    line.remove(0);
+                }
+                putln!(o, "{line}");
+            }
+            putln!(o);
+            components_and_apis(&mut o, &components, &object.fields);
+            putln!(o);
+            example_list(&mut o, &examples);
+
+            let path = self
+                .docs_dir
+                .join(format!("{}.md", object.snake_case_name()));
+            super::common::write_file(&path, &o);
+            filepaths.insert(path);
+        }
+
+        filepaths
+    }
+}
+
+fn frontmatter(o: &mut String, title: &str, order: u32) {
+    putln!(o, "---");
+    putln!(o, "title: {title}");
+    putln!(o, "order: {order}");
+    putln!(o, "---");
+}
+
+fn components_and_apis(o: &mut String, components: &[&Object], fields: &[ObjectField]) {
```

Comment:
This is unclear to the reader - why are we given a list of components, and a list of fields? Are those all the fields of all the components?

Better names can help a lot!

```suggestion
fn write_archetype_fields(o: &mut String, all_components: &[&Object], fields: &[ObjectField]) {
```